lib/resty/upload.lua

```diff
@@ -21,12 +24,52 @@ local STATE_READING_HEADER = 2
 local STATE_READING_BODY = 3
 local STATE_EOF = 4
 
+local function warped_receiveuntil(self, until_str)
```

Comment:
maybe the function name should be `wrapped_receiveuntil`

Reply:
these two local functions (this one and the `req_socket_body_collector` one) are more suitable to be placed below all the local variables (line 72)